src/core/injections/controller/controller.py

```diff
@@ -317,12 +317,9 @@ def check_parameter_in_http_header(check_parameter):
 """
 def injection_proccess(url, check_parameter, http_request_method, filename, timesec):
   settings.NOT_TESTABLE_PARAMETERS = False
-  for i in range(0,int(settings.OS_CHECKS_NUM)):
-    if settings.CHECK_BOTH_OS:
-      if i == 0:
-        settings.TARGET_OS = settings.OS.UNIX
-      else:
-        settings.TARGET_OS = settings.OS.WINDOWS
+  target_os_list = [settings.OS.UNIX, settings.OS.WINDOWS] if settings.CHECK_BOTH_OS else [settings.TARGET_OS]
+  for current_os in target_os_list:
+    settings.TARGET_OS = current_os
```

Comment:
Considerations:

- The new code simplifies the OS loop by iterating over a fixed list of `[settings.OS.UNIX, settings.OS.WINDOWS]` if `settings.CHECK_BOTH_OS` is `True`, otherwise uses `[settings.TARGET_OS]`.
- This approach improves readability and avoids explicit index checks.
- **However, it no longer respects `settings.OS_CHECKS_NUM`**, which the original code uses to control how many times the loop runs.
- If `settings.OS_CHECKS_NUM` is greater than 2, the new code may skip extra iterations that the original supported.
- Recommend verifying if `settings.OS_CHECKS_NUM` is always 2 or less when `settings.CHECK_BOTH_OS` is `True`, or else adapt the new logic to handle multiple iterations accordingly.